src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,219 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<E> {
+        private E item;
+        private Node<E> next;
+        private Node<E> prev;
+
+        Node(Node<E> prev, E element, Node<E> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (size == 0) {
+            addFirstElement(node);
+            return;
+        }
+        tail.next = node;
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
+        Node<T> newNode = new Node<>(null, value, null);
+        Node<T> oldNode = getIndexNode(index);
+        if (size == 0) {
+            addFirstElement(newNode);
+            return;
+        }
+        if (index == size || index == 0) {
+            addTopBottom(newNode, index);
+            return;
+        }
```

Comment:
can we avoid using the custom method for addFirstElement addTopBottom? For  ```index == size``` you can use method  add(T value) for another case you can use something  similar: 
```
        Node<T> next = getIndexNode(index);
        Node<T> prev = next.prev;
        Node<T> newNode = new Node<>(prev, value, next);
        if (prev == null) {
            first = newNode;
        } else {
            prev.next = newNode;
        }
        next.prev = newNode;
        size++;
```